tests/suite/test_virtual_server_configmap_keys.py

```diff
@@ -7,20 +7,22 @@
 from suite.yaml_utils import get_configmap_fields_from_yaml
 
 
-def assert_update_event_emitted(virtual_server_setup, new_list, previous_list):
+def assert_update_event_emitted(virtual_server_setup, new_list, previous_list, is_deployment):
     item_name = f"{virtual_server_setup.namespace}/{virtual_server_setup.vs_name}"
     text_valid = f"Configuration for {item_name} was updated"
     text_invalid = "was updated but was not applied"
+    expected_amount = 1 if is_deployment else 3
```

Comment:
it seems that number 3 for DS depends on the cluster - 3 means 3 nodes right? what if the number of nodes is changed? 
Should we configure that number differently instead of  hardcoding it?
Alternatively, can we get number of IC pods from the k8s API and that use that number ?

Reply:
This is a testing framework, in my opinion at each point of time test should know exactly what is going on in the system.

Reply:
@tellet 
the PR makes it is a requirement from the cluster to have 3 nodes if the tests are run with `daemon-set` deployment type. Is it possible to make it either conifgurable (so that the infrastructure can be changed without changing the source code) or at least document that in the README https://github.com/nginxinc/kubernetes-ingress/blob/master/tests/README.md ?

Reply:
I'll add a method to get # of IC pods then.